Source/drlg_l1.cpp

```diff
@@ -732,71 +732,67 @@ void ApplyShadowsPatterns()
 	}
 }
 
-bool PlaceMiniSet(const BYTE *miniset, int tmin, int tmax, int cx, int cy, bool setview)
+bool PlaceMiniSet(const BYTE *miniset, bool setview)
 {
-	int sx;
-	int sy;
-
 	int sw = miniset[0];
 	int sh = miniset[1];
+	int sx = GenerateRnd(DMAXX - sw) - 1;
+	int sy = GenerateRnd(DMAXY - sh);
 
-	int numt = 1;
-	if (tmax - tmin != 0) {
-		numt = GenerateRnd(tmax - tmin) + tmin;
-	}
-
-	for (int i = 0; i < numt; i++) {
-		sx = GenerateRnd(DMAXX - sw);
-		sy = GenerateRnd(DMAXY - sh);
-		bool abort = false;
-		int found = 0;
+	for (int bailcnt = 0;; bailcnt++) {
+		if (bailcnt > 4000)
+			return false;
 
-		while (!abort) {
-			abort = true;
-			if (cx != -1 && sx >= cx - sw && sx <= cx + 12) {
-				sx++;
-				abort = false;
-			}
-			if (cy != -1 && sy >= cy - sh && sy <= cy + 12) {
-				sy++;
-				abort = false;
+		sx++;
+		if (sx == DMAXX - sw) {
+			sx = 0;
+			sy++;
+			if (sy == DMAXY - sh) {
+				sy = 0;
 			}
+		}
 
-			int ii = 2;
+		if (SetPiecesRoom.Contains({ sx, sy })) {
+			continue;
+		}
 
-			for (int yy = 0; yy < sh && abort; yy++) {
-				for (int xx = 0; xx < sw && abort; xx++) {
-					if (miniset[ii] != 0 && dungeon[xx + sx][sy + yy] != miniset[ii])
-						abort = false;
-					if (Protected[xx + sx][sy + yy])
-						abort = false;
-					ii++;
-				}
-			}
+		bool valid = true;
+		if (sx <= 12) {
+			sx++;
+			valid = false;
+		}
+		if (sy <= 12) {
+			sy++;
+			valid = false;
+		}
+		if (!valid) {
+			continue;
+		}

```

Comment:
This code is just plain broken. It is causing candelabra and Poison Water entrance to never appear in the upper part of the map. It was probably not even intended to be activated since the input in drlg_l1 is always 0x0, but to deactivate it it should be -1x-1 like in the other levels. But we are pretty much stuck with it now.

![image](https://user-images.githubusercontent.com/204594/172889731-db6343c7-efaa-43c5-a270-d407d8ac2a6c.png)

Reply:
Wow, cool find. What happens to saves if you fix it, objects could spawn over the player and trap him? The floor torches are called "braziers" btw.

Reply:
You would get entirely different layouts so they could end up in a wall:
![image](https://user-images.githubusercontent.com/204594/173564951-66a01014-b9ad-4e56-9eb2-6c63fb5cf7ef.png)